.github/workflows/gradle.yml

```diff
@@ -9,70 +9,87 @@ on:
       - master
       - '[2-9]+.[0-9]+.x'
 jobs:
+
   build:
     runs-on: ubuntu-latest
+
     strategy:
+      fail-fast: false
       matrix:
-        java: ['11', '14']
+        java: [11, 17]
+
     env:
       WORKSPACE: ${{ github.workspace }}
       GRADLE_OPTS: -Xmx1500m -Dfile.encoding=UTF-8
+
     steps:
       - uses: actions/checkout@v4
-      - name: Set up JDK
-        uses: actions/setup-java@v4
+      - uses: gradle/wrapper-validation-action@v2
```

Comment:
I think we don't need to use gradle/wrapper-validation-action with gradle/actions/setup-gradle

Reply:
I see. But isn't it still a good protective step to prevent someone from committing a bad/malicious gradle wrapper? I can remove it if you think it's superfluous.

Reply:
@puneetbehl @matrei
FWIW I believe `gradle/actions/setup-gradle` may not be validating binaries yet. At least that's my understanding given these Gradle issues:
- https://github.com/gradle/actions/issues/12
- https://github.com/gradle/wrapper-validation-action/issues/94
- https://github.com/gradle/wrapper-validation-action/issues/173

So it might be a good idea to explicitly invoke `gradle/wrapper-validation-action` while https://github.com/gradle/actions/issues/12 is not completed.